### ISSUE TITLE
Remove support for iovisor bcc from kernel builds

### ISSUE THREAD

**Issue by rn** [linuxkit/linuxkit#3675]
We added support for building the iovisor bcc (BPF Compiler Collection) back in August 2018 and it received the last update with:
```
4dbdfb65d28b ("Move up to the 0.11.0 bcc release.")
```
on 2019-12-16. Since then it has caused quite a few build issues, esp when updating the alpine base image and I eventually had to disable it with:
```
c0f50012d21b ("kernel: Disable bcc")
```
since I was not able to fix the build.

We are trying to clean up the kernel build and reduce the number of items we build, so I'm considering removing `bcc` in about a month time.

If you are using `bcc` please let us know.

@kmjohansen tagging you here since you provided the initial support and updated it. Any thoughts?

**Comment by kmjohansen** [comment]
Hey @rn, thanks for the nudge.  I left a few comments in https://github.com/linuxkit/linuxkit/issues/3652 too.

First off, I'm sorry this has been such a pain lately.  Having tools like bcc and bpftrace available are quite useful for system-level debugging.  My general attitude about these is that it's better to have them ready, then need to sink the time into building them in the middle of the emergency that actually requires them.

I think there are a few options if we'd like to continue to provide a bcc package.

1. The build errors may be fixed in newer builds.  We could try moving up to a more recent bcc release and seeing if we get lucky and everything just works.
2. Alpine seems to be providing bcc packages from their distro now.  This seems to have started in 3.11 and has continued through 3.13.  The 3.13 release has 0.18.0 packaged, and edge contains 0.20.0.  The bcc project itself now seems to contain explicit instructions for installing both source and binary builds on Alpine.  (see https://github.com/iovisor/bcc/blob/master/INSTALL.md#alpine---source and https://github.com/iovisor/bcc/blob/master/INSTALL.md#alpine---binary respectively).  It might be easier than the first option if we included the bcc from the Alpine and pulled it into a package that way.
3. Alpine also packages libbpf, if we've reached the point where most people are transitioning to bpf CO-RE for debug tools, and are building kernels with BTF support it might make sense to drop bcc.  Though, I think most distros are supporting both right now.

**Comment by rn** [comment]
@kmjohansen thanks for the quick response. I agree that having this support ready when needed rather than scrambling to get it to work is a good approach.

However, me as a maintainer, I want to, for example upgrade the kernel or move to a newer Alpine etc. Most packages just work, but `bcc` (and `perf`) have been particularly tricky. On the latest Alpine update, I spent a reasonable amount of time trying to get `perf` and `bcc` to work (and the patches are in) but in the end I'm not an expert in building these tools. I ended up disabling the builds for `bcc` because the main objective was to update everything else.

I'd be more than happy to keep the `bcc` package there if you continue updating it as needed.

**Comment by kmjohansen** [comment]
@rn I get it, and I empathize with the frustration.  That last update I sent for bcc took a lot longer to get working than I had hoped it would too.  I never intended for this to be such a hassle, and I can understand why you're not keen on looking after it.  Apologies for that.

I don't want kernel or alpine updates to get blocked on my availability,  Perhaps it would make sense to remove this package for now, and if can get some extra time to look at this more I'll send along a new PR to bring it back to life.  At that point, we can have a discussion about whether the new PR is sustainable or just as inconvenient as the old approach.  I'm leaning towards option number two in my previous post, which seems like it might be less of a hassle.  Any thoughts about either the removal, or the options for bringing it back?

**Comment by rn** [comment]
@kmjohansen Thanks. I think I leave the code in for now but leave the build disabled as is to make it easier if you do get around to take a look.  Thanks agin for the quick response and I'll close this for now.
